docs/working-with-form.md

```diff
@@ -0,0 +1,89 @@
+
+Note that this concept only applies to the active text input element. Any elements which are marked with `data-reflex-permanent` will not be morphed in any way.
+
+## Form submission
+
+Django-sockpuppet gathers all of the attributes on the element that initiates a Reflex. All of this data gets packed into an object that is made available to your Reflex action method through the `element` accessor. You can even [scoop up the attributes of parent elements](https://sockpuppet.argpar.se/reflexes#inheriting-data-attributes-from-parent-elements). This leaves form submission in the cold, though... doesn't it? 🥶
+
+### The `params` accessor
+
+_Heck no!_ If a Reflex is called on a `form` element - or a **child** of that `form` element - then the data for the whole form will be properly serialized and made available to the Reflex action method as the `params` accessor. `params` is a dictionary that you can input into a django form as data, and then validate that data as you normally would.
+
+One of the most exciting benefits of this design is that autosaving the data in your form becomes as simple as adding `data-reflex="change->Post#update"` to each field. Since the field is inside the parent `form` element, all inputs are automatically serialized and sent to your Reflex class.
+
+You are free to add additional business logic on the client using the Reflex [lifecycle callbacks](https://sockpuppet.argpar.se/lifecycle) in your Stimulus controllers.
+
+Reflex actions called outside of a form will still have a `params` instance variable; it will be an empty dictionary.
+
+{% hint style="danger" %}
+If you call a full-page update Reflex outside of a form that has unsaved data, you will lose the data in the form. You will also lose the data if you throw your laptop into a volcano. 🌋
+{% endhint %}
+
+#### Modifying form data before sending to the server
+
+Should you need to modify the contents of your params before the Reflex sends the data to the server, you can use the `before` callbacks to do so:
+
+```javascript
+document.addEventListener('stimulus-reflex:before', event => {
+  const { params } = event.target.reflexData
+  event.target.reflexData.params = { ...params, foo: true, bar: false }
+})
+```
+
+#### A note about &lt;input type="file"&gt; fields
+
+At the time of this writing, **forms that upload files are unsupported by django-sockpuppet**. We suggest that you design your UI in such a way that files can be uploaded directly, making use of the standard django-form upload techniques. You might need to use `data-reflex-permanent` so that you don't lose UI state when a Reflex is triggered.
+
+As websockets is a text-based protocol that doesn't guarantee packet delivery or the order of packet arrival, it is not well-suited to uploading binary files. This is an example of a problem best solved with vanilla Rails.
```

Comment:
There's a reference to Rails in here when it should be Django.

Reply:
Good catch!